micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -126,6 +126,40 @@ impl NVSStorage {
     }
 }
 
+impl StorageDiagnostic for NVSStorage {
+    type Error = NVSStorageError;
+    fn log_space_diagnostic(&self) -> Result<(), Self::Error> {
+        let part_name = CString::new("nvs")
+            .map_err(|_| EspError::from_non_zero(NonZeroI32::new(ESP_ERR_INVALID_ARG).unwrap()))?;
+        let mut stats: nvs_stats_t = Default::default();
+        esp!(unsafe { nvs_get_stats(part_name.as_ptr(), &mut stats as *mut _) })?;
+
+        // 32 bytes per entry
```

Comment:
Can we link to docs that support this?

Reply:
I'm worried about posting what might be a dead link in the future